Tasks/AzureResourceGroupDeploymentV2/operations/ResourceGroup.ts

```diff
@@ -170,16 +170,71 @@ export class ResourceGroup {
             tl.error(error.message);
             if (error.details) {
                 tl.error(tl.loc("Details"));
+                let policyLink = null;
+
                 for (var i = 0; i < error.details.length; i++) {
-                    var errorMessage = util.format("%s: %s %s", error.details[i].code, error.details[i].message, error.details[i].details);
+                    var errorMessage = null;
+                    if (error.details[i].code === "RequestDisallowedByPolicy") {
+                        if (!policyLink) {
+                            policyLink = this.getPolicyHelpLink(error.details[i]);
+                        }
+
+                        errorMessage = this.getPolicyErrorMessage(error.details[i]);
+                    } else {
+                        errorMessage = util.format("%s: %s %s", error.details[i].code, error.details[i].message, error.details[i].details);
+                    }
+
                     tl.error(errorMessage);
                 }
+
+                if (policyLink) {
+                    tl.warning(tl.loc("ManageAzurePolicy") + policyLink);
+                }
             }
         } else {
             tl.error(error);
         }
     }
 
+    private getPolicyHelpLink(errorDetailMessage): string {
+        let managementGroupId = this.parseManagementGroupId(errorDetailMessage);
+        let portalUrl = this.taskParameters.endpointPortalUrl ? this.taskParameters.endpointPortalUrl : "https://portal.azure.com";
+        let policyLink = portalUrl + "#blade/Microsoft_Azure_Policy/PolicyMenuBlade/Assignments/scope/%2Fproviders%2FMicrosoft.Management%2FmanagementGroups%2F" + managementGroupId;
+        return policyLink;
+    }
+
+    private parseManagementGroupId(errorDetail): string {
+        let errorMessage = errorDetail.message;
+        let managementGroupString = "managementGroups";
+        let splitErrorMessage = errorMessage.split('/');
+
+        for (let i = 0; i < splitErrorMessage.length; i++) {
+            if (splitErrorMessage[i] === managementGroupString) {
+                managementGroupString = splitErrorMessage[i + 1];
+                break;
+            }
+        }
+
+        return managementGroupString;
+    }
+
+    private getPolicyErrorMessage(errorDetail): string {
+        var errorMessage = errorDetail.message;
+
+        if (!!errorMessage) {
+            errorMessage = errorMessage.split(".")[0] + ".";
+        }
+
+        var additionalInfo = errorDetail.additionalInfo;
+        if (!!additionalInfo) {
+            for (var i = 0; i < additionalInfo.length; i++) {
+                errorMessage += util.format(tl.loc("ErrorType") + " %s, " + tl.loc("PolicyDefinitionName") + " %s, " + tl.loc("PolicyAssignmentName") + " %s ", additionalInfo[i].type, additionalInfo[i].info.policyDefinitionDisplayName, additionalInfo[i].info.policyAssignmentName);
```

Comment:
I think the idea is not to merge + and %s. You can put tl.loc(xyz) also as another %s

Reply:
Should there be newlines or full stop between two additional info messages

Reply:
Also can additinalInfo[i].info be null? If yes can we check that before accessing it

Reply:
Adding a new line between error message doesn't print the second message in red color.  First message implicitly contains a full stop.

**Error sample**
Resource 'gbdkltdxbppxqsawinvm' was disallowed by policy. Error Type: PolicyViolation, Policy Definition Name : Define Allowed locations , Policy Assignment Name : policylocations.